qualang_tools/video_mode/README.md

```diff
@@ -0,0 +1,185 @@
+```
+param_name= new_value_1 new_value_2 new_value_3 ...
+```
+
+Once the user has typed an appropriate update, the ```VideoMode``` will use the IO variables of the ``QuantumMachine`` 
+to pass the following information:
+- which parameter of the initial dictionary should be updated next (IO1)
+- what new value should be set for the parameter (IO2)
+
+For updating a QUA array, a loop over the elements of the array is performed to update dynamically each element 
+with a successive back and forth interaction between the program and the Python (through interleaved ```pause()``` within QUA and ```job.resume()``` in Python).
+
+As everything happens behind the scenes, the user does not need to worry about the details of the implementation.
+The only thing that the user should implement in the QUA program on top of the declaration of variables shown above is the following macro:
+```
+with program() as video_mode_prog:
+    ...
+    with infinite_loop():
+        video_mode.load_parameters()
+        ...
+```
+
+### Accessing QUA variables from ```VideoMode``` within the program
+The QUA variables declared in the program through the ```VideoMode.declare_variables()``` can be accessed from the
+through the ```video_mode["param_name"]``` syntax. This enables you to choose where your QUA variable should land in 
+your program. For instance, if you want to use the same QUA variable for two different pulse parameters, you can do the following:
+```
+with program() as video_mode_prog:
+    video_mode.declare_variables()
+    
+    with infinite_loop():
+        video_mode.load_parameters()
+        
+        play("my_pulse"*amp(video_mode["amp_param"]), "qe1")
+        play("my_other_pulse"*amp(video_mode["amp_param"]), "qe2")
+```
+
+Note that you can also access to the full list of QUA variables declared in the program through the ```video_mode.variables``` attribute.
+
+Alternatively, you can also access the QUA variables later in the code as follows 
+(note that the order of the variables is the same as the order of the parameters in the `parameters_dict` as of Python 3.7):
+```
+with program() as video_mode_prog:
+    amp_param1, amp_param2, ... = video_mode.declare_variables()
+    
+    with infinite_loop():
+        video_mode.load_parameters()
+        
+        play("my_pulse"*amp(amp_param1), "qe1")
+        play("my_other_pulse"*amp(amp_param2), "qe2")
+```
+## Outside the QUA program
+The ```VideoMode``` class has only one ```execute```method to be called outside the QUA program scope, which is a simple wrapper of ```qm.execute()``` method.
+It will start the video mode execution by creating a new thread doing a continuous query of input parameters from the user, while the program is executed and possible live data plotting is on.
+Note that the QUA program runs in the background and the user can still interact with the Python environment while the program is running.
```

Comment:
This line is confusing, it sounds like the normal execute which let's you **freely** interact with python. But if I understood it correctly, you've taken over the console?

This needs rephrasing. I think this is what confused me about the examples

Reply:
Let me know if the new sentence clarifies it.

Reply:
Yes it does, thanks!